user/TASK/gimbal_task/gimbal_task.c

```diff
@@ -121,24 +172,63 @@ static void get_new_data(Gimbal_t *gimbal_data){
  */
 static void update_setpoints(Gimbal_t *gimbal_set){
     
-    gimbal_set->yaw_motor.pos_set += (-1) * int16_deadzone(gimbal_set->rc_update->rc.ch[2], -DEADBAND, DEADBAND) / 10;
-    gimbal_set->pitch_motor.pos_set += int16_deadzone(gimbal_set->rc_update->rc.ch[3], -DEADBAND, DEADBAND) / 10;
+    //yaw:
+    //rc -> theta -> complex rotation -> new setpoint
+
+    // TODO: FIX
+    if(gimbal_set->rc_update->rc.s[0] == RC_SW_MID || gimbal_set->rc_update->rc.s[0] == RC_SW_UP){
+        fp32 theta = -1 * int16_deadzone(gimbal_set->rc_update->rc.ch[2], -DEADBAND, DEADBAND)
+                * Motor_Ecd_to_Rad / 80.0f;
+        fp32 rotation[2] = {cos(theta), sin(theta)}; // TODO: consider alternative {cos{theta}, sin{theta}}
```

Comment:
See user_lib.c (whole file) and the include arm_math.h (line 4925+) for fast/stm32f4 math functions

Reply:
COOL.
Also found a PID function ~5000